### PR TITLE
[ADD] PHP-CS-Fixer

### PR DESCRIPTION
**Pull request by Th3Mouk**
### Proposition

In order to fix easily code style convention, many bundles now use [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer).

The `.php_cs` file allow configuration on your needs, actually i've include the same as [Sonata Project](https://github.com/sonata-project/SonataAdminBundle/blob/master/.php_cs).

The use is really simple, when you're developing, type `make cs` or `make cs_dry_run`, thanks to `Makefile`, he will correct all the files with defined rules.

The actual configuration manage, too automatically, headers files of the bundle.

If this solution concurs for you, you're free to modify after merge the configuration.

Cheers :beers:

### CODE REVIEW

**Comment by DavidBadura** [comment]
the change `assertEquals` to `assertSame` breaks the build.

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <?php
+
 /*
  * (c) 2011 SimpleThings GmbH
  *
  * @package SimpleThings\EntityAudit
  * @author Benjamin Eberlei <eberlei@simplethings.de>
- * @author Andrew Tch <andrew.tchircoff@gmail.com>
```
Comment (OskarStark) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: This is wrong

if we use the header, please move the author above the class name and check the other files

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php [hunk 2]
```diff
@@ -30,31 +30,33 @@
 use SimpleThings\EntityAudit\AuditConfiguration;
 use SimpleThings\EntityAudit\AuditReader;
 use SimpleThings\EntityAudit\Exception\AuditedCollectionException;
-use Traversable;
```
Comment (OskarStark) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: why is this removed?

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php [hunk 3]
```diff
@@ -514,8 +516,8 @@ protected function initialize()
 
             foreach ($rows as $row) {
                 $entity = array(
-                    'rev' => $row['rev'],
-                    'revtype' => $row['revtype']
+                    'rev'     => $row['rev'],
```
Comment (OskarStark) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: please remove the unalignment from configuration

@see https://github.com/KnpLabs/KnpPaginatorBundle/pull/338#discussion-diff-42511167

File: src/SimpleThings/EntityAudit/Exception/AuditedCollectionException.php [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <?php
+
 /*
  * (c) 2011 SimpleThings GmbH
  *
  * @package SimpleThings\EntityAudit
  * @author Benjamin Eberlei <eberlei@simplethings.de>
- * @author Andrew Tch <andrew.tchircoff@gmail.com>
```
Comment (OskarStark) [inline src/SimpleThings/EntityAudit/Exception/AuditedCollectionException.php]: same as above

File: src/SimpleThings/EntityAudit/Utils/SimpleDiff.php [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
```
Comment (OskarStark) [inline src/SimpleThings/EntityAudit/Utils/SimpleDiff.php]: not sure about this change

maybe also put the removed doc bloc above the class

**Comment by Th3Mouk** [comment]
Yes the `php_unit_strict` change the assertTest, i just removed it

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php (hunk 2 above)
Comment (Th3Mouk) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: Seems used, how is it used ?

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php (hunk 3 above)
Comment (Th3Mouk) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: Ok

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php (hunk 1 above)
Comment (Th3Mouk) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: Ok

File: src/SimpleThings/EntityAudit/Exception/AuditedCollectionException.php (hunk 4 above)
Comment (Th3Mouk) [inline src/SimpleThings/EntityAudit/Exception/AuditedCollectionException.php]: Ok

File: src/SimpleThings/EntityAudit/Collection/AuditedCollection.php (hunk 2 above)
Comment (DavidBadura) [inline src/SimpleThings/EntityAudit/Collection/AuditedCollection.php]: this is unused, is okay :)

File: src/SimpleThings/EntityAudit/Utils/SimpleDiff.php (hunk 5 above)
Comment (Th3Mouk) [inline src/SimpleThings/EntityAudit/Utils/SimpleDiff.php]: OK

**Comment by Th3Mouk** [comment]
@DavidBadura @OskarStark Any other mistakes ?

**Comment by Th3Mouk** [comment]
@DavidBadura ok for merge ?

**Comment by OskarStark** [comment]
Looks ok to me :+1:

**Comment by DavidBadura** [comment]
it looks good to me. the probleme is, after we merge this all other PR's have then merge conflict. maybe we should wait here?

**Comment by Th3Mouk** [comment]
The solution is to merged with using 'my branch' and ignoring upstream changes (if they're only relative to CS) and at the end type `make cs` command then commit =)

**Comment by OskarStark** [comment]
btw i would keep the header the same everywhere and add special authors above the class declaration. otherwise anybody will have diffs, if he execute ``make cs``...

what do you think?

**Comment by Th3Mouk** [comment]
@OskarStark in the header I put only the maintainer or co-funder, and yes special authors must be on the class declaration. The header is only relative to bundle licence. But isn't the case here in my memory

**Comment by OskarStark** [comment]
OK @Th3Mouk :+1:

File: .php_cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Symfony\CS\Fixer\Contrib\HeaderCommentFixer::setHeader($header);
+
+$finder = Symfony\CS\Finder\DefaultFinder::create()
+    ->in(array(__DIR__))
+    ->exclude(array('Tests/Fixtures'))
+;
+
+return Symfony\CS\Config\Config::create()
+    ->level(Symfony\CS\FixerInterface::SYMFONY_LEVEL)
```
Comment (smoench) [inline .php_cs]: I would just prefer the `PSR2_LEVEL`. The `SYMFONY_LEVEL` has some fixers I don't like.

File: composer.json [hunk 7]
```diff
@@ -10,7 +10,8 @@
     "require-dev": {
         "symfony/framework-bundle": "~2.3",
         "phpunit/phpunit": "~4.0",
-        "gedmo/doctrine-extensions": "~2.2"
+        "gedmo/doctrine-extensions": "~2.2",
+        "fabpot/php-cs-fixer": "^0.5|^1.0"
```
Comment (smoench) [inline composer.json]: Please use only the lastest version `^1.11`

File: README.md [hunk 8]
```diff
@@ -244,3 +244,7 @@ This provides you with a few different routes:
 * Proper metadata mapping is necessary, allow to disable versioning for fields and associations.
 * It does NOT work with Joined-Table-Inheritance (Single Table Inheritance should work, but not tested)
 * Many-To-Many associations are NOT versioned
+
+## Contributing
+
+Before each commit please run the `make cs` command.
```
Comment (smoench) [inline README.md]: Could you remove the `Makefile` and refer those commands in the `README.md` instead?

**Comment by smoench** [comment]
@Th3Mouk Could you please revert the changes from `php-cs-fixer fix`-run. I must add some changes to the header comment on my own and a git rebase would be easier. Thanks in advance.

**Comment by bendavies** [comment]
@smoench i'm going to pick this up. Do you have a preference of cs config?

**Comment by Th3Mouk** [comment]
As you want, i can modify too if necessary

**Comment by bendavies** [comment]
@Th3Mouk thanks!
if you could rebase on master and address the other comments that would great

**Comment by Th3Mouk** [comment]
@smoench @bendavies it seems good

**Comment by DavidBadura** [comment]
Thank you! :)
